src/distPlus/symbolic/SymbolicDist.re

```diff
@@ -296,103 +313,407 @@ module GenericSimple = {
+    }
+    | (`Simple(d1), `Simple(d2)) => {
+        let (sc1, sd1) = renderDistributionToXYShape(d1, n);
+        let (sc2, sd2) = renderDistributionToXYShape(d2, n);
+        `RenderedShape(Distributions.Continuous.reduce((+.), [|sc1, sc2|]), Distributions.Discrete.reduce((+.), [|sd1, sd2|]), 2.)
+    }
+    | (`Simple(d1), `RenderedShape(sc2, sd2, i2))
+    | (`RenderedShape(sc2, sd2, i2), `Simple(d1)) => {
+        let (sc1, sd1) = renderDistributionToXYShape(d1, n);
+        `RenderedShape(Distributions.Continuous.reduce((+.), [|sc1, sc2|]), Distributions.Discrete.reduce((+.), [|sd1, sd2|]), 1. +. i2)
+    }
+    | (`RenderedShape(sc1, sd1, i1), `RenderedShape(sc2, sd2, i2)) => {
+        `RenderedShape(Distributions.Continuous.reduce((+.), [|sc1, sc2|]), Distributions.Discrete.reduce((+.), [|sd1, sd2|]), i1 +. i2)
+    }
+    }
   };
-};
 
-let toString = (r: bigDist) =>
-  r
-  |> (
-    fun
+  let evaluatePointwiseProduct = (et1: nodeResult, et2: nodeResult, n: int) => {
+    switch ((et1, et2)) {
+    /* Known cases: */
+    | (`Simple(`Float(v1)), `Simple(`Float(v2))) => {
+        v1 == v2
+        ? `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.make({xs: [|v1|], ys: [|1.|]}), 1.)
+        : `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.) // TODO: add warning: shouldn't pointwise multiply scalars.
+    }
+    | (`Simple(`Float(v1)), `Simple(d2)) => {
+        // evaluate d2 at v1
+        let y = GenericSimple.pdf(v1, d2);
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.make({xs: [|v1|], ys: [|y|]}), y)
+    }
+    | (`Simple(d1), `Simple(`Float(v2))) => {
+        // evaluate d1 at v2
+        let y = GenericSimple.pdf(v2, d1);
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.make({xs: [|v2|], ys: [|y|]}), y)
+    }
+    | (`Simple(`Normal(n1)), `Simple(`Normal(n2))) => {
+        let mean = (n1.mean *. n2.stdev**2. +. n2.mean *. n1.stdev**2.) /. (n1.stdev**2. +. n2.stdev**2.);
+        let stdev = 1. /. ((1. /. n1.stdev**2.) +. (1. /. n2.stdev**2.));
+        let integral = 0; // TODO
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.)
+    }
+    /* General cases */
+    | (`Simple(d1), `Simple(d2)) => {
+        // NOT IMPLEMENTED YET
+        // TODO: evaluate integral properly
+        let (sc1, sd1) = renderDistributionToXYShape(d1, n);
+        let (sc2, sd2) = renderDistributionToXYShape(d2, n);
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.)
+    }
+    | (`Simple(d1), `RenderedShape(sc2, sd2, i2)) => {
+        // NOT IMPLEMENTED YET
+        // TODO: evaluate integral properly
+        let (sc1, sd1) = renderDistributionToXYShape(d1, n);
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.)
+    }
+    | (`RenderedShape(sc1, sd1, i1), `Simple(d1)) => {
+        // NOT IMPLEMENTED YET
```

Comment:
I think it's possible we won't implement these for a while. It would be better to throw an error, then we can catch this and do Monte Carlo instead.

Reply:
Sounds good, let's do that! I'll jump back into this in a few days and add that.

Reply:
Maybe ``nodeResult`` could also have an option for ``Error(n)``, or you could just return a Belt.Return(nodeResult, string) or similar in cases that need it.

My guess is that it's possible any node in a tree could be unparsable, so this is a node-level attribute.